### PR TITLE
Updated lesson 22

### PR DESCRIPTION
**Pull request by tzatushevskaya**

### CODE REVIEW

File: 22_networks_vlan_opnsense_vms_25-oct-2022/README.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+* Create linked clone from the manjaro template:
+  - Name: manjaro3
+  - Connect it to VLAN 20 via Proxmox UI (Network settings)
+* Check connectivity:
+  - manjaro3 to manjaro2, manjaro1, gateway, web
+  - manjaro1 to manjaro2, manjaro3
+  - manjaro2 to manjaro3, manjaro1
+---------------------------------------------------------------------------
+#### 6. Create VM for proxmox
+* Install SDN
```
Comment (irublev) [inline 22_networks_vlan_opnsense_vms_25-oct-2022/README.md:88]: where to do this? it is clear only for those who already know the answer

File: 22_networks_vlan_opnsense_vms_25-oct-2022/README.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+apt install libpve-network-perl ifupdown2 
+```  
+* Install qemu agent
+```  
+apt install qemu-guest-agent
+```  
+* Set up VLAN3 via update /etc/network/interfaces on each proxmox VM
+* Convert to template
+* Make 3 linked clones from the proxmox templates
+* Update hostname & ip address:
```
Comment (irublev) [inline 22_networks_vlan_opnsense_vms_25-oct-2022/README.md:100]: where to do this?

File: 22_networks_vlan_opnsense_vms_25-oct-2022/README.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+nano /etc/postfix/main.cf
+```
+```
+nano /etc/network/interfaces
+```
+```
+ifreload -s -a
+ifreload -a
+```
+* Re-generate certificates
```
Comment (irublev) [inline 22_networks_vlan_opnsense_vms_25-oct-2022/README.md:117]: and this?

File: 22_networks_vlan_opnsense_vms_25-oct-2022/README.md [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+systemctl restart pveproxy
+```
+* Check connectivity:
+  - each node to each node
+* Open on manjaro2 each node ui in separate tab
+* Create cluster
+
+---------------------------------------------------------------------------
+#### 7. Send cloudinit folder to each node
+* Send ssh key to each node
```
Comment (irublev) [inline 22_networks_vlan_opnsense_vms_25-oct-2022/README.md:131]: how to do this?

File: 22_networks_vlan_opnsense_vms_25-oct-2022/README.md [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+* Run script to create ubuntu cloudinit template on 1st node
```
Comment (irublev) [inline 22_networks_vlan_opnsense_vms_25-oct-2022/README.md:132]: which script?

File: 22_networks_vlan_opnsense_vms_25-oct-2022/README.md [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+  - edit a connection
+  - wired connection to know parent edit
+  - add a connection
+    - type VLAN 
+    - device vlan03
+    - parent ens18
+  - activate connection
+  - deactivate wired connection
+  - activate VLAN3
+- Check IP address:
```
Comment (irublev) [inline 22_networks_vlan_opnsense_vms_25-oct-2022/README.md:255]: where to do this?

**Review by irublev — CHANGES_REQUESTED** [review]

File: 22_networks_vlan_opnsense_vms_25-oct-2022/README.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+* Proxmox UI: System > Network > \[Create\] > Linux bridge
+  - Name
+  - VLAN aware - checked
+---------------------------------------------------------------------------
+#### 2. Create VM for OPNsense (see [Lesson 16](../16_networks_ssl-termination_self-signed_cert_04-oct-2022))
+* Set vmbr0 as WAN
+* Set vmbr1 as LAN
+  - set up LAN (subnet mask)
+* Go to OPNsense to set up the firewall:
+  - Firewall > Rules: close web to LAN (enable the rule LAN to LAN only)
```
Comment (bbs-md) [inline 22_networks_vlan_opnsense_vms_25-oct-2022/README.md:23]: - Firewall > Rules -> LAN
 web access

File: 22_networks_vlan_opnsense_vms_25-oct-2022/README.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+* Create 1st linked clone from the Manjaro template
+  - Name: manjaro1
+  - Network device: untagged VLAN (vmbr1)
+
+* Check connectivity via _ping_ command:
+  - gateway
+  - web
+
+* Show logs:
+  - Firewall > Live Log > filter by interface LAN
```
Comment (bbs-md) [inline 22_networks_vlan_opnsense_vms_25-oct-2022/README.md:35]: - Firewall > Log Files > Live View

File: 22_networks_vlan_opnsense_vms_25-oct-2022/README.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+  - Service > DHCPv4 *
+  - Firewall > Aliases > RFC
+  - Firewall > Rules > \[+\]
+* Go back to Manjaro:
+  - Open Network Manager (nmtui)
+    - edit a connection
+    - wired connection to know parent edit
+    - add a connection > vlan
+      - device vlan03
+      - parent ens18
```
Comment (bbs-md) [inline 22_networks_vlan_opnsense_vms_25-oct-2022/README.md:57]: ens18.3

File: 22_networks_vlan_opnsense_vms_25-oct-2022/README.md [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+    - Check IP address
+    ```
+    ip a
+    ```
+
+* Check connectivity:
+  - manjaro1 to manjaro2, gateway, web
+  - manjaro2 to manjaro1, gateway, web
+* Show logs:
+  - Firewall > Live Log > filter by interface
```
Comment (bbs-md) [inline 22_networks_vlan_opnsense_vms_25-oct-2022/README.md:70]: Firewall > Log Files > Live View

File: 22_networks_vlan_opnsense_vms_25-oct-2022/README.md [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+  - manjaro1 to manjaro2, manjaro3
+  - manjaro2 to manjaro3, manjaro1
+---------------------------------------------------------------------------
+- Set up network connection via _nmtui_
+  - edit a connection
+  - wired connection to know parent edit
+  - add a connection
+    - type VLAN 
+    - device vlan03
+    - parent ens18
```
Comment (bbs-md) [inline 22_networks_vlan_opnsense_vms_25-oct-2022/README.md:99]: ens18.3

File: 22_networks_vlan_opnsense_vms_25-oct-2022/README.md [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+  - Firewall > Rules > \[+\]
+* Create linked clone from the Manjaro template:
+  - Name: manjaro3
+  - Connect it to VLAN 20 via Proxmox UI (Network settings)
+* Check connectivity:
+  - manjaro3 to manjaro2, manjaro1, gateway, web
+  - manjaro1 to manjaro2, manjaro3
+  - manjaro2 to manjaro3, manjaro1
+---------------------------------------------------------------------------
+- Set up network connection via _nmtui_
```
Comment (bbs-md) [inline 22_networks_vlan_opnsense_vms_25-oct-2022/README.md:93]: In `manjaro3` VM terminal

**Pull request merged**